flink-connector/flink/src/main/java/org/apache/gravitino/flink/connector/PropertiesConverter.java

```diff
@@ -32,25 +34,80 @@ public interface PropertiesConverter {
 
   /**
    * Converts properties from application provided properties and Flink connector properties to
-   * Gravitino properties.
+   * Gravitino properties.This method processes the Flink configuration and transforms it into a
+   * format suitable for the Gravitino catalog.
    *
-   * @param flinkConf The configuration provided by Flink.
-   * @return properties for the Gravitino catalog.
+   * @param flinkConf The Flink configuration containing connector properties. This includes both
+   *     Flink-specific properties and any user-provided properties.
+   * @return A map of properties converted for use in the Gravitino catalog. The returned map
+   *     includes both directly transformed properties and bypass properties prefixed with {@link
+   *     #FLINK_PROPERTY_PREFIX}.
    */
   default Map<String, String> toGravitinoCatalogProperties(Configuration flinkConf) {
-    return flinkConf.toMap();
+    Map<String, String> gravitinoProperties = Maps.newHashMap();
+    for (Map.Entry<String, String> entry : flinkConf.toMap().entrySet()) {
+      String gravitinoKey = transformPropertyToGravitinoCatalog(entry.getKey());
+      if (gravitinoKey != null) {
+        gravitinoProperties.put(gravitinoKey, entry.getValue());
+      } else if (entry.getKey().startsWith(FLINK_PROPERTY_PREFIX)) {
+        gravitinoProperties.put(entry.getKey(), entry.getValue());
+      }
```

Comment:
if user create a catalog with properties not defined by gravitino or start with `flink.bypass.`,  we should add `flink.bypass.` to it.

Reply:
If that's the case, I don't need to make this modification